src/FFT/FFT.h

```diff
@@ -251,6 +274,25 @@ namespace ippl {
          */
         void transform(TransformDirection direction, Field& f);
     };
+    /**
+       Cosine type 1 transform class
+    */
+    template <typename Field>
+    class FFT<Cos1Transform, Field> : public IN_PLACE_FFT_BASE_CLASS(Field, backendCos1) {
+        constexpr static unsigned Dim = Field::dim;
+        using Base                    = IN_PLACE_FFT_BASE_CLASS(Field, backendCos1);
+
+    public:
+        using Base::Base;
+        using typename Base::heffteBackend, typename Base::workspace_t, typename Base::Layout_t;
+
+        /*!
+         * Perform in-place FFT
```

Comment:
What do you mean by "in-place FFT"?

Reply:
Output data is saved in input view (as opposed to r2c and c2r which need to have different input and output destinations because of data mismatch).